### PR TITLE
Remove deprecated apt_key ansible module

### LINKED ISSUE #1004: apt-key is deprecated and scheduled for removal after ubuntu 22.04
We are making use of `apt-key` for managing APT keys. This tool is deprecated and scheduled for removal after 22.04 (so we likely need to fix this before upgrading to 24.04).

References:

- [man page for apt-key](http://manpages.ubuntu.com/manpages/jammy/man8/apt-key.8.html). See the deprecation notice. There is a suggestion for how to manage the keys without apt-key.
- [the ansible module we are using](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/apt_key_module.html#notes)

### PR DESCRIPTION
**Pull request by Sunnatillo**
fixes #1004

### CODE REVIEW

**Comment by Sunnatillo** [comment]
//test-ubuntu-integration-e2e-main

**Review by lentzi90 — COMMENTED** [review]
/approve

**Review by furkatgofurov7 — CHANGES_REQUESTED** [review]

File: vm-setup/roles/packages_installation/tasks/ubuntu_required_packages.yml [hunk 1]
```diff
@@ -6,21 +6,31 @@
         name: "*"
         state: latest
 
-    - name: Add OS release key
+    # TODO: Sunnatillo Remove this task after fully removing apt-key
```
[the remark this answers is not in the document]
Comment (Sunnatillo) [reply vm-setup/roles/packages_installation/tasks/ubuntu_required_packages.yml]: Done

File: vm-setup/roles/packages_installation/tasks/ubuntu_required_packages.yml [hunk 2]
```diff
@@ -46,13 +56,31 @@
...
+            dest: /etc/apt/keyrings/docker.asc
+            mode: '0644'
+            force: true
+          
+        - name: Dearmor GPG key
+          shell: | 
+            cat /etc/apt/keyrings/docker.asc | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg --yes
+                
+        # TODO: (Sunnatillo) Remove this task after fully removing apt-key
+        - name: Remove Docker old repository(without gpg key file location)
```
Comment (tuminoid) [inline vm-setup/roles/packages_installation/tasks/ubuntu_required_packages.yml:70]: Nit: add space between `repository` and `(` for readability.

File: vm-setup/roles/packages_installation/tasks/ubuntu_required_packages.yml [hunk 3]
```diff
@@ -46,13 +56,31 @@
...
+            cat /etc/apt/keyrings/docker.asc | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg --yes
+                
+        # TODO: (Sunnatillo) Remove this task after fully removing apt-key
+        - name: Remove Docker old repository(without gpg key file location) 
+          apt_repository:
+            repo: "deb [arch=amd64] https://download.docker.com/linux/ubuntu {{ ansible_distribution_release }} stable"
+            state: absent
+
+        # TODO: (Sunnatillo) Remove this task after fully removing apt-key
+        - name: Remove Docker old repository(without gpg key file location)
```
Comment (tuminoid) [inline vm-setup/roles/packages_installation/tasks/ubuntu_required_packages.yml:76]: Nit: add space between `repository` and `(` for readability.

File: vm-setup/roles/packages_installation/tasks/ubuntu_required_packages.yml [hunk 4]
```diff
@@ -6,21 +6,31 @@
         name: "*"
         state: latest
 
-    - name: Add OS release key
+    # TODO: (Sunnatillo) Remove this task after fully removing apt-key
+    - name: Remove OS old repository(without gpg key file location)
```
Comment (tuminoid) [inline vm-setup/roles/packages_installation/tasks/ubuntu_required_packages.yml:10]: Nit: add space between `repository` and `(` for readability.

**Review by tuminoid — COMMENTED** [review]
Just nits.

File: vm-setup/roles/packages_installation/tasks/ubuntu_required_packages.yml [hunk 5]
```diff
@@ -46,13 +56,31 @@
...
+          shell: | 
+            cat /etc/apt/keyrings/docker.asc | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg --yes
+                
+        # TODO: (Sunnatillo) Remove this task after fully removing apt-key
+        - name: Remove Docker old repository (without gpg key file location) 
+          apt_repository:
+            repo: "deb [arch=amd64] https://download.docker.com/linux/ubuntu {{ ansible_distribution_release }} stable"
+            state: absent
+
+        # TODO: (Sunnatillo) Remove this task after fully removing apt-key
```
Comment (furkatgofurov7) [inline vm-setup/roles/packages_installation/tasks/ubuntu_required_packages.yml:75]: is `apt-key` still used in other places? what does fully removing mean here?

**Review by furkatgofurov7 — APPROVED** [review]
/lgtm

File: vm-setup/roles/packages_installation/tasks/ubuntu_required_packages.yml (hunk 5 above)
Comment (Sunnatillo) [reply vm-setup/roles/packages_installation/tasks/ubuntu_required_packages.yml:75]: We have images that have the repo keys which was managed by apt-key. Once we have new images without apt-key managed keys, then conflicting part is gone. We can remove these tasks.

File: vm-setup/roles/packages_installation/tasks/ubuntu_required_packages.yml (hunk 4 above)
Comment (Sunnatillo) [reply vm-setup/roles/packages_installation/tasks/ubuntu_required_packages.yml]: done

File: vm-setup/roles/packages_installation/tasks/ubuntu_required_packages.yml (hunk 3 above)
Comment (Sunnatillo) [reply vm-setup/roles/packages_installation/tasks/ubuntu_required_packages.yml]: done

File: vm-setup/roles/packages_installation/tasks/ubuntu_required_packages.yml (hunk 2 above)
Comment (Sunnatillo) [reply vm-setup/roles/packages_installation/tasks/ubuntu_required_packages.yml]: done

File: vm-setup/roles/packages_installation/tasks/ubuntu_required_packages.yml (hunk 5 above)
Comment (furkatgofurov7) [reply vm-setup/roles/packages_installation/tasks/ubuntu_required_packages.yml:75]: Thanks, SGTM

**Review by furkatgofurov7 — COMMENTED** [review]
/lgtm

**Pull request merged**